tensorflow_datasets/core/download/kaggle.py

```diff
@@ -46,10 +47,44 @@
 class KaggleFile(object):
   """Represents a Kaggle competition file."""
   _URL_PREFIX = "kaggle://"
+  kaggle_type = collections.namedtuple(
+      "KaggleType",
+      ["prefix", "download_cmd", "dl_flag", "extra_flag"])
+  _DATASET_TYPE = kaggle_type(
+      prefix="dataset",
+      download_cmd="datasets",
+      dl_flag="-d",
+      extra_flag="--unzip")
+  _COMPETITION_TYPE = kaggle_type(
+      prefix="competition",
+      download_cmd="competitions",
+      dl_flag="-c",
+      extra_flag="")
+  KAGGLE_TYPES = {
+      _COMPETITION_TYPE.prefix: _COMPETITION_TYPE,
+      _DATASET_TYPE.prefix: _DATASET_TYPE
+      }
 
   def __init__(self, competition_name, filename):
     self._competition_name = competition_name
     self._filename = filename
+    self._type = None
+
+  @staticmethod
+  def get_type(competition_name):
```

Comment:
I would make this a simple function instead and move `KAGGLE_TYPES` outside the class.

Reply:
I don't see the changes ? did you forgot to push ?

I was talking about removing `@staticmethod` and moving this outside the class.

```
def _get_type(competition_name):
  ...
```

Reply:
Oh Shoot! Forgot to fix that. Thanks for reminding. :)